hatch_pip_compile/lock.py

```diff
@@ -44,24 +45,28 @@ def process_lock(self) -> None:
         """
         prefix = dedent(raw_prefix).strip()
         joined_dependencies = "\n".join([f"# - {dep}" for dep in self.dependencies])
-        lockfile_text = self.lock_file.read_text()
+        lockfile_text = lockfile.read_text()
         cleaned_input_file = re.sub(
             rf"-r \S*/{self.env_name}\.in",
             f"hatch.envs.{self.env_name}",
             lockfile_text,
         )
         if self.constraints_file is not None:
+            constraint_sha = hashlib.sha256(self.constraints_file.read_bytes()).hexdigest()
             constraints_path = self.constraints_file.relative_to(self.project_root)
-            constraints_line = f"# [constraints] {constraints_path}"
-            joined_dependencies = "\n".join([constraints_line, "#", joined_dependencies])
+            constraints_lines = [
+                f"# [constraints]: {constraints_path}",
+                f"# [constraints-sha]: {constraint_sha}",
```

Comment:
Good feature. I was just thinking, should the design of these headers converge on being "generic" like this (one per line) or perhaps something more "cute" and "integrated", e.g.

```
[constraints] requirements.txt (SHA256: 8f0f4a2dddff0e6f0499b14b933e53eccfd7678d11339b395b044b97c348e390)
```

Reply:
I like this better

```
# [constraints] requirements.txt (SHA256: 8f0f4a2dddff0e6f0499b14b933e53eccfd7678d11339b395b044b97c348e390)
```

Reply:
Just curious, how have you been staying up to date as more upgrades are pushed to the plugin? I haven't adopted it in other projects yet.

If you install the plugin like this, is there an easy way to update? Something like `hatch env run --env=internal pip install --upgrade hatch-pip-compile` would be nice

```toml
[tool.hatch.env]
requires = [
    "hatch-pip-compile"
]
```

Reply:
1. I actually use local checkouts of many repos and add all of them to PYTHONPATH so it kicks in instantly after any pull or change (can't recommend though, it's the most volatile approach)

2. Generally how to upgrade - I guess just run the same pip as was used to install hatch? Or pipx: `pipx inject --pip-args=-U hatch hatch-pip-compile`

3. Or you could probably add a version constraint to the `requires` in a project?